### ISSUE TITLE
Android Face Detection using CameraX draws rectangles inaccurately around faces

### ISSUE THREAD

**Issue by MenaRaafat** [googlesamples/mlkit#90]
When using the CameraX live preview option with face detection mode, the detected faces are surrounded by rectangles inaccurately as shown in the below screenshot:
![Screenshot_20200722-153606](https://user-images.githubusercontent.com/42410083/88183207-75cc2b80-cc31-11ea-9f0c-c1d54fa54875.png)

This is present in both the Java and Kotlin samples.

This issue is not present when using the live preview without CameraX option as shown in the below screenshot:
![Screenshot_20200722-153547](https://user-images.githubusercontent.com/42410083/88183457-c6dc1f80-cc31-11ea-8aa1-def398a0f465.png)

Is this issue caused by a bug in the CameraX library, or a bug in the sample?

**Comment by ya332** [comment]
Can you experiment with the face min width ratio in the settings when you run the application?

**Comment by calren** [comment]
Hey @MenaRaafat , thanks for filing this! We're not aware of any related bugs in the CameraX library , so it sounds like it might be a bug with the sample. We'll look into this more and keep you updated.

**Comment by MenaRaafat** [comment]
Fix found.
Setting the constraints of the **preview_view** in **activity_camerax_live_preview.xml** to: wrap content, and binding it to all sides, fixes the issue.

**Comment by steven-chengji-yan** [comment]
Hi @MenaRaafat , would you mind tell us the OS version you are using when saw this issue?
It works fine on my device with API 28 and 29.

**Comment by MenaRaafat** [comment]
@steven-chengji-yan  I am using Nougat, 7.0, API level 24 on Huawei Mate 8. Change the resolution maybe it triggers the bug.

**Comment by steven-chengji-yan** [comment]
Hi MenaRaafat, I tried on Android N with a Nexus 5X, but still cannot reproduce it. 

I saw in your screenshot, the resolution is 400x400 and 408x544, which the camera on my Nexus 5X doesn't have.

I am glad you found a working solution for you. However that actually changed the behavior a little bit. Previously, the camera preview takes the entire area above the bottom bar. With your fix, there are more black padding now.    When creating this app, we intentionally apply centerCrop to the preview to make sure it covers the entire area that we want to show the preview, even if the View area and camera preview have different aspect ratio.

Will leave the code as is for now and investigate more if see other reports.

Thanks for filing this!
